main.go

```diff
@@ -462,3 +473,62 @@ func stopStalePipelines (client *gitlab.Client, vars []*gitlab.PipelineVariable)
 
 	}
 }
+
+// syncEnterpriseRepos detects whether a commit has been merged to
+// the Open Source edition of a repo, and then sends a GET request to
+// the Gallienii server, asking it to synchronize the repos
+func syncEnterpriseRepos(conf *config, gpr *github.PullRequestEvent) error {
+
+	repo := gpr.GetRepo()
+	if repo == nil {
+		return fmt.Errorf("syncEnterpriseRepos: Failed to get the repository information")
+	}
+
+	// Enterprise repo sentinel
+	switch repo.GetName() {
+	case "deployments":
+	case "useradm":
+	default:
+		// Do not run sync, whenever there is no merge to an Enterprise repo
+		return nil
+	}
+
+	pr := gpr.GetPullRequest()
+	if pr == nil {
+		return fmt.Errorf("syncEnterpriseRepos: Failed to get the pull request")
+	}
+
+	// Only sync on Merges to master
+	branch := pr.GetBase()
+	if branch == nil {
+		return fmt.Errorf("syncEnterpriseRepos: Failed to get the base branch of the PR: %v", branch)
+	}
+	branchRef := branch.GetRef()
+	if branchRef != "master" {
+		log.Debugf("syncEnterpriseRepos: Detected a merge into another branch than master: (%s), do not sync", branchRef)
+		return nil
+	}
```

Comment:
Sorry to bi picky... but this message here is wrong: at this point, you don't know if the action is a merge or any other activity (pushing new commits).

You should move this if case inside `if gpr.GetAction() == "closed" && pr.GetMerged()` or just extend this if with one condition more

Reply:
Lol, yes. Good point!